crates/bench/src/game.rs

```diff
@@ -0,0 +1,198 @@
+        ("dimension", AlgebraicType::U32),
+    ];
+
+    let indexes = &[
+        (0.into(), "entity_id_index"),
+        (1.into(), "chunk_index"),
+        (2.into(), "x_index"),
+    ];
+    let location_state = create_table(&db, &mut tx, "LocationState", schema, indexes)?;
+
+    // FootprintTileState
+    let schema = &[
+        ("entity_id", AlgebraicType::U64),
+        ("type", AlgebraicType::I32),
+        ("owner_entity_id", AlgebraicType::U64),
+    ];
+
+    let indexes = &[(0.into(), "entity_id_index")];
+    let footprint_tile_state = create_table(&db, &mut tx, "FootprintTileState", schema, indexes)?;
+
+    db.commit_tx(&ExecutionContext::default(), tx)?;
+
+    Ok(Tables {
+        db,
+        _tmp: tmp,
+        location_state,
+        footprint_tile_state,
+    })
+}
+
+fn fill_tables(tables: &Tables, data: GameData) -> Result<(), DBError> {
+    let db = &tables.db;
+    let mut tx = db.begin_mut_tx();
+
+    for i in 0..data.location_state {
+        for chunk_id in 0u64..100 {
+            db.insert(
+                &mut tx,
+                tables.location_state,
+                product![i, chunk_id, (i + 10) as i32, (i + 20) as i32, (i + 30) as u32],
+            )?;
+        }
+    }
+
+    for i in 0..data.footprint_tile_state {
+        db.insert(
+            &mut tx,
+            tables.footprint_tile_state,
+            product![i, (i + 10) as i32, i + 30],
+        )?;
+    }
+
+    db.commit_tx(&ExecutionContext::default(), tx)?;
+
+    Ok(())
+}
+
+pub fn query(tables: &Tables, data: GameData) -> Result<usize, DBError> {
+    let db = &tables.db;
+    let tx = db.begin_tx();
+    let mut found = 0;
+    for chunk_index in 0..data.footprint_tile_state {
+        found += run(
+            db,
+            &format!(
+                "SELECT FootprintTileState.*
+FROM FootprintTileState
+JOIN LocationState ON entity_id = LocationState.entity_id
+WHERE LocationState.chunk_index = {chunk_index}"
+            ),
+            AuthCtx::for_testing(),
+        )?[0]
+            .data
+            .len();
+    }
+
+    db.release_tx(&ExecutionContext::default(), tx);
+
+    Ok(found)
+}
+
+pub fn game_insert(base_rows: usize) -> ResultTest<Duration> {
+    let tables = make_tables()?;
+
+    let data = GameData::new(base_rows as u64);
+
+    let start = Instant::now();
+    black_box(fill_tables(&tables, data)?);
+
+    let result = black_box(run(&tables.db, "SELECT * FROM LocationState", AuthCtx::for_testing())?);
+
+    assert_eq!(result[0].data.len(), EXPECT_ROWS);
+
+    Ok(start.elapsed())
+}
+
+pub fn game_query(base_rows: usize) -> ResultTest<Duration> {
+    let tables = make_tables()?;
+
+    let data = GameData::new(base_rows as u64);
+    fill_tables(&tables, data)?;
+
+    let start = Instant::now();
+
+    let result = query(&tables, data)?;
+
+    #[cfg(debug_assertions)]
+    assert_eq!(result, EXPECT_ROWS / 100);
+
+    #[cfg(not(debug_assertions))]
+    assert_eq!(result, EXPECT_ROWS / 10);
```

Comment:
Slightly confused about the math here, could we define these as constants next to EXPECT_ROWS?